spotinst/resource_spotinst_elastigroup_aws_suspend_processes.go

```diff
@@ -0,0 +1,283 @@
+		return err
+	}
+
+	log.Printf("===> Elastigroup read successfully: %s <===", resourceData.Id())
+	return nil
+}
+
+////-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
+////            Create
+////-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
+
+func resourceSpotinstAWSSuspendProcessesCreate(resourceData *schema.ResourceData, meta interface{}) error {
+
+	log.Printf(string(commons.ResourceOnCreate), commons.SuspendProcessesResource.GetName())
+
+	suspendProcesses, err := commons.SuspendProcessesResource.OnCreate(resourceData, meta)
+	if err != nil {
+		return err
+	}
+
+	SuspendProcessesId, err := createSuspendProcesses(resourceData, suspendProcesses, meta.(*Client))
+	if err != nil {
+		return err
+	}
+
+	resourceData.SetId(spotinst.StringValue(SuspendProcessesId))
+
+	log.Printf("===> SuspendProcesses created successfully for Elastigroup: %s <===", resourceData.Id())
+
+	return resourceSpotinstAWSSuspendProcessesRead(resourceData, meta)
+
+}
+
+func createSuspendProcesses(resourceData *schema.ResourceData, suspendProcesses *aws.SuspendProcesses, spotinstClient *Client) (*string, error) {
+	if json, err := commons.ToJson(suspendProcesses); err != nil {
+		return nil, err
+	} else {
+		log.Printf("===> SuspendProcesses create configuration: %s", json)
+	}
+	input := &aws.CreateSuspensionsInput{Suspensions: suspendProcesses.Suspensions}
+	input.GroupID = spotinst.String(resourceData.Get(string(elastigroup_aws_suspend_processes.GroupID)).(string))
+
+	err := resource.Retry(time.Minute, func() *resource.RetryError {
+		_, err := spotinstClient.elastigroup.CloudProviderAWS().CreateSuspensions(context.Background(), input)
+		if err != nil {
+			// an error occurred, no retryable errors for this resource.
+			return resource.NonRetryableError(err)
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, fmt.Errorf("[ERROR] failed to create SuspendProcesses for Elastigroup: %s", err)
+	}
+	return input.GroupID, nil
+
+}
+
+//-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
+//           Delete
+//-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
+func resourceSpotinstAWSSuspendProcessesDelete(resourceData *schema.ResourceData, meta interface{}) error {
+	resourceId := resourceData.Id()
+	log.Printf(string(commons.ResourceOnDelete), commons.SuspendProcessesResource.GetName(), resourceId)
+
+	if err := deleteSuspendProcesses(resourceData, meta); err != nil {
+		return err
+	}
+
+	log.Printf("===> SuspendProcesses deleted successfully for Elastigroup: %s <===", resourceData.Id())
+	resourceData.SetId("")
+	return nil
+}
+
+func deleteSuspendProcesses(resourceData *schema.ResourceData, meta interface{}) error {
+
+	listInput := &aws.ListSuspensionsInput{}
+	gID := resourceData.Id()
+	listInput.GroupID = &gID
+
+	curr, err := meta.(*Client).elastigroup.CloudProviderAWS().ListSuspensions(context.Background(), listInput)
+	if err != nil {
+		return fmt.Errorf("[ERROR] failed to update suspend processes")
```

Comment:
Please expose the upstream error.

```suggestion
		return fmt.Errorf("[ERROR] failed to update suspend processes: %v", err)
```